src/pages/indicators/Merge.tsx

```diff
@@ -21,18 +21,15 @@ export const Component = () => {
     const location = useLocationWithState<{ selectedModels: Set<string> }>()
 
     const dataEngine = useDataEngine()
-    const initialValues = useMemo(() => {
-        const defaults = {
-            ...getDefaults(mergeFormSchema),
-            target: undefined,
-            sources: Array.from(location.state?.selectedModels || []).map(
-                (id) => ({
-                    id,
-                })
-            ),
-        }
-        return defaults
-    }, [location.state?.selectedModels])
+    const selectedIds: string[] = useMemo(
+        () => Array.from(location.state?.selectedModels ?? []),
+        [location.state?.selectedModels]
+    )
+    const initialValues = {
+        ...getDefaults(mergeFormSchema),
+        target: undefined,
+        sources: [],
+    }
```

Comment:
I think it's preferable that the initial values are stable. If it re-renders at this level it may reinitalize the form.